web/src/stream_popover.js

```diff
@@ -562,6 +564,64 @@ export async function build_move_topic_to_stream_popover(
         $("#move_topic_form .move_messages_edit_topic").trigger("focus");
     }
 
+    // The following logic is correct only when
+    // both message_lists.current.data.fetch_status.has_found_newest
+    // and message_lists.current.data.fetch_status.has_found_oldest are true;
+    // otherwise, we cannot be certain of the correct count.
+    function get_count_of_messages_to_be_moved(selected_option) {
+        if (selected_option === "change_one") {
+            return 1;
+        }
+        if (selected_option === "change_later") {
+            return message_util.get_count_of_messages_in_topic_sent_after_current_message(
+                current_stream_id,
+                topic_name,
+                message.id,
+            );
+        }
+        return message_util.get_messages_in_topic(current_stream_id, topic_name).length;
+    }
+
+    function update_move_messages_count_text(selected_option) {
+        const terms = narrow_state.filter().terms();
+        const message_move_count = get_count_of_messages_to_be_moved(selected_option);
+        const has_found_newest = message_lists.current.data.fetch_status.has_found_newest();
+        const has_found_oldest = message_lists.current.data.fetch_status.has_found_oldest();
+        const narrowed_stream_id = terms[0].operand;
+        const narrowed_topic_name = terms[1].operand;
+        const is_topic_narrowed = narrow_state.narrowed_by_topic_reply();
+        const is_stream_narrowed = narrow_state.narrowed_by_stream_reply();
+        const is_same_stream = narrowed_stream_id === current_stream_id;
+        const is_same_topic = narrowed_topic_name === topic_name;
+
+        let message_text;
+        if (
+            (is_same_stream || is_same_topic) &&
```

Comment:
Don't think this logic is correct. Like for a case where stream is same but it is narrowed to a different topic, then the `message_lists.current` will not contain data of the topic from whihch message is being moved.

Please just go through the cases once and make sure that the logic is correct.